### PR TITLE
Properly abort uploads

### LINKED ISSUE #14897: An unknown error occurred when canceling big upload
### Steps to reproduce
1. Upload a large file (>2 GB)
2. Cancel the upload while it is still running
3. Error message pop up

### Expected behaviour
Tell us what should happen: Confirmation (Upload cancelled by User) or nothing

### Actual behaviour
Tell us what happens instead: An unknown error occured. In the log I see 

Error | PHP | rmdir(/media/nextclouddata/admin/uploads/web-file-upload-18d7f2204cc78a95bddd48c88affbadc-1553760974752): No such file or directory at /var/www/nextcloud/lib/private/Files/Storage/Local.php#118
-- | -- | --

### Server configuration

**Operating system**: Ubuntu Server 18.04  LTS

**Web server:** Apache2

**Database:** MariaDB

**PHP version:** 7.2

**Nextcloud version:** (see Nextcloud admin page):15.05.03

**Updated from an older Nextcloud/ownCloud or fresh install:**fresh install

**Where did you install Nextcloud from:** Nextcloud download page

**Signing status:**
<details>
<summary>Signing status</summary>

```
Login as admin user into your Nextcloud and access 
http://example.com/index.php/settings/integrity/failed 
paste the results here.
No errors have been found.
</details>

**List of activated apps:**
<details>
<summary>App list</summary>

Standard apps as after a fresh install

If you have access to your command line run e.g.:
sudo -u www-data php occ app:list
from within your Nextcloud installation folder
```
</details>

**Nextcloud configuration:**
<details>
<summary>Config report</summary>

```
If you have access to your command line run e.g.:
sudo -u www-data php occ config:list system
from within your Nextcloud installation folder

or 

Insert your config.php content here. 
Make sure to remove all sensitive content such as passwords. (e.g. database password, passwordsalt, secret, smtp password, …)
```
</details>

**Are you using external storage, if yes which one:** local/smb/sftp/...
no
**Are you using encryption:** yes/no
yes
**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...
no

### Client configuration
**Browser:**
Google Chrome Ver. 73
**Operating system:**
Win 10
#### Web server error log
<details>
<summary>Web server error log</summary>

```
Insert your webserver log here
```
</details>

#### Nextcloud log (data/nextcloud.log)
<details>
<summary>Nextcloud log</summary>

```
Insert your Nextcloud log here

Error | PHP | rmdir(/media/nextclouddata/admin/uploads/web-file-upload-18d7f2204cc78a95bddd48c88affbadc-1553760974752): No such file or directory at /var/www/nextcloud/lib/private/Files/Storage/Local.php#118 |  
-- | -- | -- | --



</details>

#### Browser log
<details>
<summary>Browser log</summary>

```
Insert your browser log here, this could for example include:

a) The javascript console log
b) The network log
c) ...
```
</details>

### PR DESCRIPTION
**Pull request by PVince81**
Add a new approach for flagging an upload as aborted because we can't
rely on the browser fully cancelling the request as we now seem to
receive an error response from the server instead of a jQuery "abort"
message.

Fixes: https://github.com/nextcloud/server/issues/14897

- [x] TEST: abort one download
- [x] TEST: abort multiple download
- [x] TEST: test error message (404 when upload on non-existing folder) ~~progress bar not removed!~~
- [x] TEST: upload files again after an error

### CODE REVIEW

**Comment by PVince81** [comment]
since now the deletion of the upload object is delayed, it messed up with the accounting that operates the progress bar

I've pushed a fix to properly count the finished uploads so that the progress bar can be hidden, also in error cases.

File: apps/files/js/file-upload.js [hunk 1]
```diff
@@ -322,26 +322,34 @@ OC.FileUpload.prototype = {
...
+		this.deleteUpload();
+	},
+
 	/**
 	 * Abort the upload
 	 */
 	abort: function() {
-		if (this.data.isChunked) {
-			this._deleteChunkFolder();
+		var self = this;
```
Comment (artonge) [inline apps/files/js/file-upload.js:337]: self variable looks unused

**Review by artonge — APPROVED** [review]
Looks good.

**Review by Pytal — APPROVED** [review]

File: apps/files/js/file-upload.js (hunk 1 above)
Comment (PVince81) [reply apps/files/js/file-upload.js:337]: ah, leftover from previous fiddling. thanks for noticing

**Comment by PVince81** [comment]
I've fixed some bogus references to self and retested

**Comment by GuyPaddock** [comment]
This is excellent news!! Thank you so much, @PVince81!

**Pull request merged**